adrs/20241029-versioning.md

```diff
@@ -0,0 +1,68 @@
+# Versioning Scheme Tied to FAST Releases
+
+**authors:** [Ludo](https://github.com/ludoo),  [Julio](https://github.com/jccb), [Simone](https://github.com/sruffilli) \
+**date:** Oct 29, 2024
+
+## Status
+
+Proposed
+
+## Context
+
+Our current versioning scheme releases new versions based on changes across all modules. This approach was suitable when modules were the primary focus of development. However, with the increasing importance of FAST, this process no longer aligns with our priorities. We need a versioning scheme that reflects the significance of FAST releases and allows for more frequent updates to modules and documentation. The current release process wasn't designed with FAST in mind, causing friction and delaying releases.
+
+## Proposal
+
+Tie major version releases to FAST releases containing breaking changes or new architectural paradigms. Minor version releases will be used for module updates and documentation changes.
+
+### Development Workflow:
+
+* **Modules and Documentation:** Changes to modules and documentation will be made directly to the `master` branch.
```

Comment:
Can we add here: `FAST bugfixes` and (optionally) `FAST minor changes`. 

I agree that with minor changes we blur the line, but bugfixes should be pretty straightforward.

Reply:
what we discussed last night is something like

- is the change impacting the user by requiring a change to tfvars/inputs, or renames in state, or removing functionality? ---> fast-dev/rc release process/major version
- is the change impacting modules' interfaces or state? ---> master/minor version (including any changes to FAST to adapt to the interface, modulo above statement)
- anything else ---> master/patch release or wait for next release

Reply:
some additional notes

- it is the developer's responsibility to merge master in fast-dev via the branch they're developing on (eventually merged in fast-dev via PR)
- we always do back-to-back releases when doing rc releases, to avoid having master changes in the rc changelog